### DISCUSSION TITLE
Static Setup File possible?

### DISCUSSION THREAD

**Discussion by Zelsof** [Bodmer/TFT_eSPI#2904] [Compatible displays and setup files]
Hi, 

Everytime there is an update for TFT_eSPI availabe, after install, my "User_Setup.h" has been overwritten. Is there an easy way to keep this file static? Or to include the parameters like "#define TFT_CS ... " somehow in my code?

The aim would be to update the library without caring about the setup file.

Ideas? :)

**Comment by Bodmer** [comment]
If you use PlatformIO the setup can be in the platformio.ini file, see here:
https://github.com/Bodmer/TFT_eSPI/blob/master/docs/PlatformIO/Configuring%20options.txt

You can have your own folder for setups and link the library to it, see end of tips here:
https://github.com/Bodmer/TFT_eSPI#tips

The key advantage of the setup file approach is that examples will run without modification.

**Reply by Zelsof** [reply]
wow, that's amazing. Excactly what I needed. I am using this library already such a long time and I didn't notice that this is now possible.

I am using PlatformIO and everything works now as it should. Thank you sir!
